### ISSUE TITLE
Filtering trees by Any Tag Set returns duplicate records

### ISSUE THREAD

**Issue by nmcharlton** [Greenstand/treetracker-admin#609]
Passing the following query to GET /trees returns duplicate records:
```
filter: "where":{"approved":true,"active":true,"tagId":"0"}
```

Trees (captures) appear to be included once per tag that they have applied, e.g., a tree with three tags appears three times in the results.

**Comment by nmcharlton** [comment]
@bstetzer32 Would you mind looking at this one?

**Comment by nmcharlton** [comment]
Fix reverted – see #612

**Comment by ZavenArra** [comment]
I think this issue requires the specification of which column names( necessary to return and distinct) 

SELECT DISTINCT id, (other column names) FROM ...

Though actually, it seems to me that the below solution (which was reverted) should work.
```
        const columnNames = this.dataSource.connector
          .buildColumnNames('Trees', filter)
          .replace('"id"', 'trees.id as "id"');

const selectStmt = `SELECT DISTINCT ${columnNames} from trees ${this.getTreeTagJoinClause(
          tagId,
        )}`;

```
OR use of the DISTINCT ON clause.

https://www.postgresql.org/docs/current/sql-select.html
https://www.geekytidbits.com/postgres-distinct-on/
